src/Nri/Ui/Table/V2.elm

```diff
@@ -0,0 +1,281 @@
+module Nri.Ui.Table.V2
+    exposing
+        ( Column
+        , custom
+        , keyframeStyles
+        , keyframes
+        , string
+        , view
+        , viewLoading
+        , viewLoadingWithoutHeader
+        , viewWithoutHeader
+        )
+
+{-|
+
+@docs Column, custom, string
+
+@docs view, viewWithoutHeader
+
+@docs viewLoading, viewLoadingWithoutHeader
+
+@docs keyframes, keyframeStyles
+
+-}
+
+import Css exposing (..)
+import Html.Styled as Html exposing (..)
+import Html.Styled.Attributes exposing (css)
+import Nri.Ui.Colors.V1 exposing (..)
+import Nri.Ui.Fonts.V1 exposing (baseFont)
+import Nri.Ui.Styles.V1
+
+
+{-| Closed representation of how to render the header and cells of a column
+in the table
+-}
+type Column data msg
+    = Column (Html msg) (data -> Html msg) Style
+
+
+type alias WidthStyle =
+    List ( String, String )
+
+
+{-| A column that renders some aspect of a value as text
+-}
+string :
+    { header : String
+    , value : data -> String
+    , width : LengthOrAuto compatible
```

Comment:
> Is there a danger in allowing relative column sizes I'm overlooking?

Not really a danger, but people might find that they now want to set `table-layout: fixed` on the table element to, perhaps tweak the layout to have a column auto-expand while keeping the widths of the rest fixed (["Example 3: Mixed unit widths" of this article](https://adamschwartz.co/magic-of-css/chapters/3-tables/)) - which I don't think is a need that should be addressed now.

Reply:
Can see the use case for this, but going to hold of on this for now!